content/en/blog/_posts/2024-06-20-pod-failure-policy-ga

```diff
@@ -0,0 +1,124 @@
+of running computational workloads, by:
+- letting you fail the Job as soon as you observe a non-retriable Pod failure
+- using small `.backoffLimit` by ignoring Pod failures due to infrastructure
+  failures (e.g. [graceful node shutdown](/docs/concepts/cluster-administration/node-shutdown/#graceful-node-shutdown) of a spot node)
+
+## How can you use it? {#pod-failure-policy-how-to-use}
+
+Pod failure policy is part of the Pod template specification a user can supply
+when creating a Job.
+
+It is represented as an ordered list of rules evaluated in order against the
+end state of a Pod. Each rule describes the match condition, which can be
+based on:
+- container exit codes (using the `onExitCodes` property),
+- Pod conditions (using the `onPodConditions` property).
+
+Each rule specifies an action (`action` property) to take in case of a match.
+The possible values include:
+- `Ignore` - don't count the failure towards the `.backoffLimit`
+- `FailJob` - fail the Job and terminate all running Pods
+- `FailIndex` - fail the index corresponding to the failed Pod (can only be
+   used with Indexed Jobs and [Backoff limit per index](/docs/concepts/workloads/controllers/job/#backoff-limit-per-index))
+- `Count` - the default way of handling a Pod failure - count towards the
+   (`.backoffLimit` or `.backoffLimitPerIndex` if specified)
+
+### Pod conditions
+
+A significant part of the development work was for this feature was to introduce
+the built-in `DisruptionTarget` condition to annotate reasons for Pod failures
+due to disruptions. The condition can have the following values of the `reason`
+field, depending on the scenario:
+
+- `PreemptionByKubeScheduler` - Pod preempted by kube-scheduler
+- `DeletionByTaintManager` - Pod evicted by kube-controller-manager due to taints
+- `EvictionByEvictionAPI` - Pod deleted by Eviction API
+- `DeletionByPodGC` - an orphaned Pod deleted by pod GC
+- `TerminationByKubelet` - Pod terminated due to graceful node shutdown or preemption for critical pods.
+
+Notably, Pod failure policy for Jobs can be used against custom user-supplied
+conditions, either added by webhooks or external controllers.
+
+Users may use them to annotate for:
+- annotating user-specific Pod failures, such as configuration issues,
+- annotating Kubernetes failure scenarios which are currently not yet recognized
+  by any built-in conditions (e.g. admission failures due to `OutOfPods`).
+
+### Example
+
+The Pod failure policy snippet below demonstrates how to use to use it to:
+- ignore any Pod disruption recognized by the built-in DisruptionTarget condition
+- fail the Job in case of configuration issues annotated by the custom `ConfigIssue`
+- fail the Job in case of Pod failure with the reserved exit code 42
+- count all other failures towards the default `backoffLimit` (or
+  `backoffLimitPerIndex` if used)
+
+```yaml
+podFailurePolicy:
+  rules:
+  - action: Ignore
```

Comment:
IMO it would be good to see two examples, one with `onExitCodes` and one with `onPodConditions`. And then show a demo how this works with Pod state progressions.

Even though we have a similar thing in https://kubernetes.io/docs/tasks/job/pod-failure-policy/, we can give slightly different examples to give more exposure to the feature. @mimowo thoughts?

We can also  mention the evaluation order and that multiple rules can be used somewhere.

Reply:
I think that that type of information is better in a documentation page tbh, and not in a blog. One to two smaller examples in blog is fine but the demo and full working examples should be in docs, which should be linked to from the blog :)

Reply:
I agree. I think it makes sense to refer to the "hands on guide" we already have as the first link in "Lean more". 

Instead, I tentatively split the example into 3 smaller ones to also include backoff limit per index. Let me know your view.

Reply:
Ok, we can make it shorter and send people to the docs instead.